app/src/main/java/com/github/se/icebreakrr/ui/map/HeatMap.kt

```diff
@@ -68,131 +63,132 @@ private val GRADIENT_START_POINTS =
+                      .maxIntensity(15.0)
+                      .build()
+            } else {
+              // Optionally handle the case where there are no valid locations
+              Log.w("HeatMap", "No valid locations to display on the heatmap.")
             }
-
-            GoogleMap(
-                modifier = Modifier.fillMaxSize().padding(paddingValues).testTag("googleMap"),
-                cameraPositionState = cameraPositionState,
-                onMapLoaded = {
-                    isMapLoaded = true // Set map loaded state to true
-                    val center = cameraPositionState.position.target
-                    profilesViewModel.getFilteredProfilesInRadius(
-                        center = GeoPoint(center.latitude, center.longitude),
-                        radiusInMeters = RADIUS // Adjust radius as needed
-                    )
-                }
-            ) {
-                // Only show heatmap if we have data
+          }
+
+          GoogleMap(
+              modifier = Modifier.fillMaxSize().padding(paddingValues).testTag("googleMap"),
+              cameraPositionState = cameraPositionState,
+              onMapLoaded = {
+                isMapLoaded = true
+                val center = cameraPositionState.position.target
+                profilesViewModel.getFilteredProfilesInRadius(
+                    center = GeoPoint(center.latitude, center.longitude), radiusInMeters = RADIUS)
+                lastCameraPosition = center
+              }) {
                 heatmapProvider?.let { provider ->
-                    TileOverlay(tileProvider = provider, transparency = 0.0f)
+                  TileOverlay(tileProvider = provider, transparency = 0.0f)
                 }
-            }
-
-            // Fetch profiles when the camera position changes
-            LaunchedEffect(cameraPositionState.position) {
-                if (isMapLoaded) {
-                    val center = cameraPositionState.position.target
+              }
+
+          // Fetch profiles when the camera position changes
+          LaunchedEffect(cameraPositionState.position) {
+            if (isMapLoaded) {
+              val center = cameraPositionState.position.target
+              val lastPosition = lastCameraPosition ?: center
+
+              // Ensure lastPosition is valid before creating Location objects
+              if (lastPosition.latitude.isNaN() || lastPosition.longitude.isNaN()) {
+                return@LaunchedEffect
+              }
+
+              // Create Location objects for distance calculation
+              val currentLocation =
+                  Location("").apply {
+                    latitude = center.latitude
+                    longitude = center.longitude
+                  }
+
+              val previousLocation =
+                  Location("").apply {
+                    latitude = lastPosition.latitude
+                    longitude = lastPosition.longitude
+                  }
+
```

Comment:
This means that if you do only small movements on the map, it would technically never reload. I guess it's not that important but still a little weird haha

Reply:
Changed it no worries